src/Validation/SubscriptionValidator.php

```diff
@@ -22,13 +22,12 @@ class SubscriptionValidator {
 	use CanValidateField;
 
 	public const SOURCE_EMAIL = 'email';
-	public const SOURCE_TITLE = 'title';
 
-	private $mailValidator;
-	private $duplicateValidator;
-	private $textPolicyValidator;
-	private $textPolicyViolations;
-	private $titleValidator;
+	private EmailValidator $mailValidator;
+	private SubscriptionDuplicateValidator $duplicateValidator;
+	private TextPolicyValidator $textPolicyValidator;
+	private array $textPolicyViolations;
+	private AllowedValuesValidator $titleValidator;
```

Comment:
You can also drop the the `$textPolicyValidator`, `$titleValidator` and `$textPolicyViolations`, I'll take care of removing those in the frontend.